src/main/java/com/algolia/search/AnalyticsConfig.java

```diff
@@ -1,44 +1,57 @@
 package com.algolia.search;
 
 import com.algolia.search.models.common.CallType;
-import java.util.Collections;
-import java.util.EnumSet;
-import java.util.List;
+import java.util.*;
 import javax.annotation.Nonnull;
 
-@SuppressWarnings({"WeakerAccess", "unused"})
+@SuppressWarnings({"unused"})
 public final class AnalyticsConfig extends ConfigBase {
 
-  /**
-   * Creates an {@link AnalyticsConfig} with the default host
-   *
-   * @param applicationID The Application ID
-   * @param apiKey The API Key
-   */
-  public AnalyticsConfig(String applicationID, String apiKey) {
-    super(applicationID, apiKey);
+  public static class Builder extends ConfigBase.Builder<Builder> {
 
-    List<StatefulHost> hosts =
-        Collections.singletonList(
-            new StatefulHost("analytics.algolia.com", EnumSet.of(CallType.READ, CallType.WRITE)));
+    /**
+     * Creates an {@link AnalyticsConfig} with the default host
+     *
+     * @param applicationID The Application ID
+     * @param apiKey The API Key
+     */
+    public Builder(@Nonnull String applicationID, @Nonnull String apiKey) {
+      super(applicationID, apiKey);
 
-    this.setDefaultHosts(hosts);
+      List<StatefulHost> hosts =
+          Collections.singletonList(
+              new StatefulHost("analytics.algolia.com", EnumSet.of(CallType.READ, CallType.WRITE)));
+
+      this.setDefaultHosts(hosts);
+    }
+
+    /**
+     * Creates a {@link AnalyticsConfig} for the search client with custom {@link StatefulHost}.
+     * Warning: Defaults hosts are not set when setting custom {@link StatefulHost}.
```

Comment:
(It would be awesome to find an API making what is happening - overriding default hosts - clear)

Reply:
Wait: why not just have a "hosts" list, defaulting to, well, default hosts, with a "setHosts" method? This would make it very clear that default hosts would get overridden.

Reply:
Nice! If we are on the same page, we would like to drop the default builder constructor: 

```java
    SearchConfig config =
        new SearchConfig.Builder("appID", "apiKEY", Collections.singletonList(new StatefulHost("Algolia", EnumSet.of(CallType.READ))))
            .build();
```

to the classic builder pattern. I like it :D WDYT?

```java
    SearchConfig config =
        new SearchConfig.Builder("appID", "apiKEY")
            .setCustomHosts(
                Collections.singletonList(new StatefulHost("Algolia", EnumSet.of(CallType.READ))))
            .build();
```

Reply:
Or with `setHosts` instead of `setCustomHosts`

Reply:
Proposal: c38029d

Reply:
Yes, that's it: `setHosts`, and Builders' constructors taking only applicationID and apiKey 👍

Reply:
Perfect! Thanks :)